status-im/status-react
# 🗝️ Password reset UI

### Summary
Implement password reset functionality:
- UI: https://www.figma.com/file/dEIljL7UPbXgsZUA0Q4qlE5E/Onboarding?node-id=8101%3A3945
- Companion PR: https://github.com/status-im/status-go/pull/2223


### Testing notes
1. Test if reset password works
   - Go to privacy > reset password
   - Enter info
   - Logout
   - You should not be able to login with old password
   - You should be able to login with new password

2. Test if delete account works with the new password
   - After resetting password, goto the delete profile screen
   - You should not be able to delete profile with old password
   - You should be able to delete profile with the new password

3. Test if password reset throws errors if wrong `current password` is supplied
4. Test if the app works fine just after password is reset :
  - Password reset doesn't logs the user out
  - But I don't know if it has any side-effects
  - Try to create chats or change settings (perform any action that writes to DB) and make sure it works 

#### Platforms

- Android
- iOS

#### Areas that maybe impacted
- Login
- Profile deletion
- Keycard migration
- Anything that makes a user enter a password (except for registration)

##### Functional
- account recovery
- new account
- user profile updates

### Steps to test
- Goto profile
- Privacy settings
- Reset password form
